r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs

```diff
@@ -633,13 +641,13 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
         //     2 * bits_per_limb + surfeit' <= CAPACITY - 2,
         // ``
         // with `surfeit' = log(num_limbs * (num_add(L) + 1) * (num_add(R) + 1) + num_limbs)`.
-        let params = get_params(SimulationF::size_in_bits(), ConstraintF::size_in_bits());
-        let num_add_bound =  BigUint::from(params.num_limbs) 
+        //ToDo: check this constraint, as by comparison with prereduce constraint a factor of 2 seems to be missing
+        let num_add_bound =  BigUint::from(params.num_limbs)
             * (BigUint::one() + &self.num_of_additions_over_normal_form)
             * (BigUint::one() + &other.num_of_additions_over_normal_form);
 
         let surfeit_prime = ceil_log_2!(
-            &num_add_bound + BigUint::from(params.num_limbs) 
+            BigUint::from(2usize)*(BigUint::one()+&num_add_bound) + BigUint::from(params.num_limbs)
```

Comment:
As far as I remember, the `num_add_bound` corresponds to `num adds + 1`.  Hence we should have 
```
let surfeit_prime = ceil_log_2!(
            BigUint::from(2usize) * &num_add_bound + BigUint::from(params.num_limbs) 
)
```
Or am I missing here something?

Reply:
Yeah, I think you are right, I will fix it, thanks